lib/components/src/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,207 @@
+import type { AddToMapFn, OpenMap } from './AccordionContext';
+import { AccordionContext } from './AccordionContext';
+
+type AccordionMap = Record<string, { index: number; id: string }>;
+
+interface StateChange {
+  id: string;
+  index: number;
+}
+
+export type AccordionProps = {
+  /** Set to true to allow for more than one AccordionItem open at the time */
+  allowMultipleOpen?: boolean;
+  /** Ability to control which panel(s) are open or closed externally - or use to set default opens */
+  open?: number | number[];
+  /** Set to true to add theme border around the Accordion */
+  bordered?: boolean;
+  /** Set to true to add rounded borders from theme to Accordion */
+  rounded?: boolean;
+  /**
+   * Set to true to add theme borders between AccordionItem's
+   * Note: can be overridden on AccordionItem
+   */
+  lined?: boolean;
+  /**
+   * Set to true to indent AccordionBody content left to match the AccordionHeader text
+   * Note: can be overridden on AccordionItem
+   */
+  indentBody?: boolean;
+  /**
+   * Set to true to apply narrow paddings and smaller text in all AccordionItem's
+   * Note: can be overridden on AccordionItem
+   */
+  narrow?: boolean;
+  /**
+   * Set to true to prevent toggle between open and close for all AccordionItem's
+   * Note: can be overridden on AccordionItem
+   */
+  preventToggle?: boolean;
+  /** Courtesy callback prop for when a new accordion opens */
+  onOpen?: (change: StateChange) => void;
+  /** Courtesy callback prop for when an accordion closes */
+  onClose?: (change: StateChange) => void;
+} & React.HTMLAttributes<HTMLUListElement>;
+
+/**
+ *
+ * Controlling wrapper component that will control the open/close functionality to
+ * child ```<AccordionItem />```'s and style the other support components
+ * ```<AccordionHeader />``` and ```<AccordionBody />```
+ *
+ * Although each component can work independently and be controlled from the outside
+ * via an **open** prop, it is not recommended to use ```<AccordionItem />```,
+ *  ```<AccordionHeader />``` and ```<AccordionBody />``` without this controller
+ *
+ * If no ```<AccordionBody />``` is inside an ```<AccordionItem />``` the open/close
+ * functionality will be removed by default, as well as the icon and pointer cursor.
+ */
+export const Accordion: FC<AccordionProps> = ({
+  allowMultipleOpen,
+  children,
+  open,
+  rounded = false,
+  bordered = false,
+  narrow, // To be carried to AccordionItem children via context
+  lined, // To be carried to AccordionItem children via context
+  indentBody, // To be carried to AccordionItem children via context
+  preventToggle, // To be carried to AccordionItem children via context
+  onOpen,
+  onClose,
+  ...rest
+}) => {
+  const [openState, setOpenState] = useState<OpenMap>({});
+  const openStateMap = useRef<OpenMap>({});
+  const accordionMap = useRef<AccordionMap>({});
+  const accordionMapIndex = useRef(0);
```

Comment:
Can you explain what each of these do?
what is openState vs openStateMap vs accordionMap?

Reply:
#### openState
`openState` is a record of id's with a boolean to tell the open/closed state of each `AccordionItem` item registered

```ts
{ "accordionItem1": false, "accordionItem2": true }
```

Simply used to control the open/closed state of items via useState

#### openStateMap
`openStateMap` is what `openState` is built on, but has to be a ref to allow each `AccordionItem` to add itself to the map before setting `openState` via useState - so we only get the one render when each `AccordionItem` has been added.

#### accordionMap
`accordionMap` is a ref of the `AccordionItem`'s in the Accordion - an object for each with the `id` and `index` (position in accordion)
```ts
{ "accordionItem1": { "id": "accordionItem1", index: 1}, "accordionItem2": { "id": "accordionItem2", index: 1}}
```